specs/agents/tracing-instrumentation-aws-lambda.md

```diff
@@ -0,0 +1,127 @@
+# AWS Lambda instrumentation
+This spec specifies the instrumentation of applications / services running on AWS Lambda.
+
+An AWS Lambda application needs to implement a **handler method** that is called whenever that Lambda function is invoked. A handler method receives (at least) two objects with useful information:
+- `event`: Depending on the trigger type of the Lambda function this object contains different, trigger-specific data. For some trigger types this object might be even missing / being a void object.
+- `context`: This object provides generic (trigger agnostic) meta information about the Lambda function.
+
+In our instrumentation we use these objects to derive useful meta and context information.
+
+## Generic Lambda Instrumentation
```

Comment:
Question: Should we consider capturing the AWS ARN of the invoked function (`context.invokedFunctionArn`) and the service that invoked the event? (`event.Records[n].eventSourceARN`).

Reply:
Added `context.invokedFunctionArn` to `faas.id`, though I'm not sure that `context.invokedFunctionArn` must be always the same. Read somewhere that there can be aliases for the same function.

`event.Records[n].eventSourceARN` is only applicable for event-based triggers. Added this in the corresponding sections for SQS, SNS, S3